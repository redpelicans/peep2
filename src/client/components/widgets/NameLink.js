import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Colors } from '@blueprintjs/core';

export const NameLink = styled(Link)`
  text-overflow: ellipsis;
  overflow: hidden;
  text-transform: capitalize;
  font-size: 0.9rem;
  font-weight: bold;
  color: ${Colors.LIGHT_GRAY5} !important;
  text-decoration: none !important;
`;

export default NameLink;
