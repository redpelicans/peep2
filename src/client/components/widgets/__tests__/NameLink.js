import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme'; // eslint-disable-line
import NameLink from '../NameLink';

describe('<NameLink />', () => {
  test('should render an <NameLink />', () => {
    const tree = shallow(<NameLink to="" />);
    expect(tree).toMatchSnapshot();
  });
  test('should <NameLink /> have text transform', () => {
    const tree = shallow(<NameLink to="" />);
    expect(tree).toHaveStyleRule('text-transform', 'capitalize');
  });
});
