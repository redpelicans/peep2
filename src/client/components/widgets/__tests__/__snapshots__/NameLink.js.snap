// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameLink /> should render an <NameLink /> 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Link
    className="sc-bdVaJa foEiOL"
    replace={false}
    to=""
/>,
  "nodes": Array [
    <Link
      className="sc-bdVaJa foEiOL"
      replace={false}
      to=""
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Styled(Link)
      to=""
/>,
    "_instance": StyledComponent {
      "attrs": Object {},
      "context": Object {},
      "props": Object {
        "to": "",
      },
      "refs": Object {},
      "state": Object {
        "generatedClassName": "foEiOL",
        "theme": null,
      },
      "unsubscribeId": -1,
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": Object {
      "generatedClassName": "foEiOL",
      "theme": null,
    },
    "_rendered": <Link
      className="sc-bdVaJa foEiOL"
      replace={false}
      to=""
/>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Styled(Link)
    to=""
/>,
}
`;
